challenge3.py

```diff
@@ -0,0 +1,51 @@
+"""
+   Advent of Code 2018 Day 3
+"""
+from itertools import product, chain, groupby
+import re
+
+from common.input_file import get_transformed_input
+
+def get_claim(claim):
+    """
+        Transform a claim to an ID and a list of points
+    """
+    claim_id, left, top, width, height = re.split(r"@|,|:|x", claim)
+    left, top, width, height = int(left), int(top), int(width), int(height)
+    points = product(range(left, left + width), range(top, top + height))
+    return (claim_id, list(points))
+
+def get_claimed_squares(claims):
+    """
+        Return a dictionary of all claimed squares where key is the coordinate
+        and the value is the number of claims
+    """
+    all_points =  chain.from_iterable(points for _, points in claims)
+    rearranged_points = groupby(sorted(all_points))
+    return {point: len(list(claims)) for point, claims in rearranged_points}
+
+def get_number_of_overlapping_squares(claimed_squares):
+    """
+        Get the number of overlapping squares
+    """
+    return (len([_ for _, claims in claimed_squares.items() if claims > 1]))
```

Comment:
If you put _, does it just put None in that entry of the list?

Reply:
it actually puts the point (As that is the key of claimed_squares of items) and that's bound to _.  _ has no syntactic special meaning in Python, it's just all semantics.  I am using it to indicate a "Don't really care" value